queue_services/business-filer/src/business_filer/services/document_service.py

```diff
@@ -0,0 +1,59 @@
+import copy
+from flask import current_app
+
+from business_model.models import Filing, UserRoles
+from document_record_service import (
+    DocumentRecordService, 
+    RequestInfo as DrsRequestInfo, 
+    get_document_class,
+    get_document_type
+)
+
+from business_filer.services import Flags
+from business_filer.services.publish_event import PublishEvent
+
+def sync_drs(filing_submission: Filing, flags: Flags): # noqa: PLR0915, PLR0912
+    document_id_state = filing_submission.filing_json["filing"]["header"].get("documentIdState", {})
+    legal_type = filing_submission.filing_json["filing"]["business"].get("legalType")
+    submitter_roles = filing_submission.submitter_roles
+
+    if  document_id_state and flags.is_on("enable-document-records"):
+        filing_type = filing_submission.filing_json["filing"]["header"]["name"]
+        temp_reg = filing_submission.temp_reg
+        
+        # Replace temp_reg with business_identifier for static documents(or staff filing)
+        if (filing_type in ["incorporationApplication", "continuationIn"]
+            or (
+                submitter_roles == UserRoles.staff
+                and filing_type
+                and document_id_state["valid"]
+                and temp_reg
+        )):
+            # Get existing document on DRS
+            doc_list = DocumentRecordService().get_document(
+                DrsRequestInfo(
+                    document_class=get_document_class(legal_type),
+                    consumer_identifier=temp_reg
+                )
+            )
+
+            if not isinstance(doc_list, list):
+                current_app.logger.error(
+                    f"No associated documents found for temporary registration ID: {temp_reg}"
+                )
+            else:
+                # Update missing consumer document id
```

Comment:
Can you explain this scenario?

Reply:
I'm not sure if this will happen, but it's an edge case where the document service ID could be missing.